### PR TITLE
Added more color configuration options

### LINKED ISSUE #587: Add .fg.* color options for every .bg.* option.
For example, there's `statusbar.bg.*` but no `fg.*` options.

### PR DESCRIPTION
**Pull request by ProtractorNinja**
Mostly resolves #587, but also makes a few additions & modifications to setting documentation. Pylint doesn't like my function to get similar configuration options: See my comment on 7b5d2ace2485b91707ce03b79e76be6580c51d41.

### CODE REVIEW

**Comment by The-Compiler** [comment]
Did a quick look at the pylint issue - either I'm misunderstanding something, or pylint is certain it could infer this as an empty string, which makes no sense at all...

```
(Pdb) sect_arg
<Const(str) l.366 [qutebrowser.browser.downloads] at 0x7ff1bc2821d0>
(Pdb) sect_arg.value
'colors'
(Pdb) opt_arg
<Const(str) l.8 [builtins] at 0x7ff1d44e6400>
(Pdb) opt_arg.value
''
```

**Comment by The-Compiler** [comment]
I brought the pylint issue up in the `#pylint` IRC channel and it turns out to be a [pylint/astroid bug](https://bitbucket.org/logilab/astroid/issue/104/).

Please add a comment like this after the docstring in `get_status_color`:

```python
# pylint: disable=bad-config-call
# WORKAROUND for https://bitbucket.org/logilab/astroid/issue/104/
```

(`WORKAROUND` is a string I use for workarounds for upstream bugs, so I can find and remove them at some point)

Then pylint should pass successfully.

As for reviewing, it'll probably take me a few days as there are some other PRs open and I also have some little things I started and want to push first - stay tuned!

**Comment by ProtractorNinja** [comment]
Those lines fixed the linter -- ready for further review.

File: qutebrowser/mainwindow/statusbar/bar.py [hunk 1]
```diff
@@ -104,14 +110,26 @@ class StatusBar(QWidget):
...
 
-        QWidget#StatusBar[insert_active="true"] {
-            {{ color['statusbar.bg.insert'] }}
+        QWidget#StatusBar QLabel {
+            {{ color['statusbar.fg'] }}
+        }
+
+        QWidget#StatusBar QLineEdit {
+            {{ color['statusbar.fg'] }}
+        }
```
Comment (The-Compiler) [inline qutebrowser/mainwindow/statusbar/bar.py]: Can't you combine those two to say `QWidget#StatusBar QLabel QLineEdit {` (like the deleted one at the bottom of the stylesheet)?

File: qutebrowser/mainwindow/statusbar/bar.py [hunk 2]
```diff
@@ -124,14 +142,54 @@ class StatusBar(QWidget):
...
+        QWidget#StatusBar[command_active="true"] QLineEdit {
+            {{ color['statusbar.fg.command'] }}
+        }
+
         QLabel, QLineEdit {
-            {{ color['statusbar.fg'] }}
             {{ font['statusbar'] }}
         }
+
     """
```
Comment (The-Compiler) [inline qutebrowser/mainwindow/statusbar/bar.py]: For some entries here you can combine them too (like mentioned above) - also in some cases there is no `QLineEdit` entry, only a `QLabel` one - is that intentional?

File: qutebrowser/mainwindow/statusbar/bar.py [hunk 3]
```diff
@@ -264,6 +322,21 @@ def _set_prompt_active(self, val):
         self.setStyleSheet(style.get_stylesheet(self.STYLESHEET))
 
     @pyqtProperty(bool)
+    def command_active(self):
+        """Getter for self.command_active, so it can be used as Qt property."""
+        return self._command_active
+
+    def _set_command_active(self, val):
```
Comment (The-Compiler) [inline qutebrowser/mainwindow/statusbar/bar.py]: This doesn't seem to be used anywhere - isn't it superseded by `set_mode_active`?

Comment (ProtractorNinja) [inline qutebrowser/mainwindow/statusbar/bar.py]: Good catch; I'll remove it.

File: qutebrowser/mainwindow/statusbar/bar.py (hunk 2 above)
Comment (ProtractorNinja) [inline qutebrowser/mainwindow/statusbar/bar.py]: Sort of -- I figured that certain modes would never see one or the other, but it's confusing to write the stylesheet that way. I'm cleaning it up.

File: qutebrowser/mainwindow/statusbar/bar.py (hunk 1 above)
Comment (ProtractorNinja) [inline qutebrowser/mainwindow/statusbar/bar.py]: Yep, I'm working on improving the messy blob of a stylesheet.

**Comment by ProtractorNinja** [comment]
I've addressed the extra setter and the messy stylesheet -- ready for more review.

**Comment by The-Compiler** [comment]
The only thing missing here is a config migration in `RENAMED_OPTIONS` in `config/config.py`.

I think something like `('colors', 'downloads.fg'): 'downloads.fg.start'` should work - but please test this (i.e. set the color to something custom on master, switch to your branch, restart, make sure download foreground still has this color).

**Comment by ProtractorNinja** [comment]
I added a patch that looks to me like it works. Tox tests are all still good on my end, too.

**Comment by The-Compiler** [comment]
Thanks! You can the AppVeyor status - I'm experimenting with that in #747 and it seems it even builds branches without an `.appveyor.yml` config.

I'll test and merge later today.

**Pull request merged**